### DISCUSSION TITLE
Change "concurrent" value dynamically in mergeMap operator

### DISCUSSION THREAD

**Discussion by pnsvk** [ReactiveX/rxjs#6249] [Ideas / Feature request]
Realising that there is no way to actually modify the mergeMap operator's 'concurrency' value dynamically, after the stream starts flowing..

Can you please let me know if this is something that can be supported out of the box by some means ?

My use-case is to let the end-users play with the number of concurrent API-Fetch calls to be made dynamically.. just as a typical download-manager would let a user dynamically change the number of concurrent downloads from a huge list of queued up download-links..

Not just this, but it actually has a larger impact overall on our application-suite..

FYI, there was exactly the same request on reactive-dotnet long time back, but was closed stating no valid reason for its closure.

https://github.com/dotnet/reactive/issues/48

**Comment by roddc** [comment]
Yes, I would like to change the concurrency through the stream
